filesystem/replicator_test.go

```diff
@@ -0,0 +1,121 @@
+package filesystem
+
+import (
+	"os"
+	"testing"
+	"time"
+)
+
+func TestReplicateProcessFileCallBack(t *testing.T) {
+	t.Run("NonExistentSourceFile", func(t *testing.T) {
```

Comment:
`NonExistentSourceFile` Shall we have a proper test title? What the test case is about?

Reply:
Same goes for other test cases as well.